crates/flux-infer/src/fixpoint_encoding.rs

```diff
@@ -1269,31 +1270,26 @@ impl<'genv, 'tcx> ExprEncodingCtxt<'genv, 'tcx> {
             .name
     }
 
-    fn register_rust_const(&mut self, def_id: DefId) -> fixpoint::GlobalVar {
+    fn register_rust_const(
+        &mut self,
+        def_id: DefId,
+        scx: &mut SortEncodingCtxt,
+        info: &rty::ConstantInfo,
+    ) -> fixpoint::GlobalVar {
         let key = Key::Const(def_id);
         self.const_map
             .entry(key)
             .or_insert_with(|| {
-                // TODO(nilehmann) generalize const sorts
-                let ty = self.genv.tcx().type_of(def_id).no_bound_vars().unwrap();
-                assert!(ty.is_integral());
-
-                // FIXME(nilehmann) we should probably report an error in case const evaluation
-                // fails instead of silently ignore it.
-                let val = self
-                    .genv
-                    .tcx()
-                    .const_eval_poly(def_id)
-                    .ok()
-                    .and_then(|val| {
-                        let val = val.try_to_scalar_int()?;
-                        rty::Constant::from_scalar_int(self.genv.tcx(), val, &ty)
-                    });
-
+                // generate and insert to `const_map`
+                let val = match info {
+                    rty::ConstantInfo::Uninterpreted => None,
+                    rty::ConstantInfo::Interpreted(expr) => Some(expr.clone()),
+                };
+                let sort = self.genv.sort_of_def_id(def_id).unwrap().unwrap();
```

Comment:
Maybe the code would look cleaner if we only record the `DefId` here and then deal with everything in `assume_const_values` by calling `genv.conf_info` and `genv.sort_of_def_id`. But I'm conflicted, I also like the idea of having the `rty::ConstantInfo` stored inside `ExprKind::ConstDefId`.

Reply:
Yes, I was not clear as to which is better, so lets leave this as is for now till a clearer picture emerges..